### PR TITLE
try to dedupe API function serialization code

### PR DESCRIPTION
**Pull request by Connoropolous**
I still have to test this to see if calling `env["hc_debug"]` works or not, but if so, sweet because we would really benefit to not have all that duplicated code

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -50,46 +49,30 @@ export const enum ErrorCode {
   PageOverflowError = 8, // returned by hdk if offset+size exceeds a page
 }
 
+function handleSerialization(input: string, api_call: string): string {
+  let encoded_allocation: u32 = serialize(input);
+  let result: u32 = env[api_call](encoded_allocation);
```
Comment (willemolding) [inline index.ts]: Assemblyscript doesn't support accessing object members using indexing syntax :(

File: index.ts [hunk 2]
```diff
@@ -49,9 +49,9 @@ export const enum ErrorCode {
   PageOverflowError = 8, // returned by hdk if offset+size exceeds a page
 }
 
-function handleSerialization(input: string, api_call: string): string {
+function handleSerialization(input: string, api_call: (e: u32) => u32): string {
```
Comment (willemolding) [inline index.ts]: Adds a function signature type as the parameter

File: index.ts [hunk 3]
```diff
@@ -62,17 +62,17 @@ function handleSerialization(input: string, api_call: string): string {
 }
 
 export function debug(message: string): void {
-  return handleSerialization(message, "hc_debug");
+  handleSerialization(message, (e: u32): u32 => env.hc_debug(e));
```
Comment (willemolding) [inline index.ts]: For some reason you can't just pass env.hc_debug it has to be wrapped in an inline function. It's kinda messy but maybe better than all that duplication

Comment (Connoropolous) [reply index.ts]: I thought you couldn't pass functions as params at all, so despite the oddities, this is still good news

**Pull request merged**